sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -676,19 +676,20 @@ case class WholeStageCodegenExec(child: SparkPlan)(val codegenStageId: Int)
           case _: WholeStageCodegenExec =>
           case _: InputRDDCodegen =>
           case c: CodegenSupport =>
-            val (newReusableExpressions, newAttributeSet) = c.reusableExpressions()
+            val (newReusableExpressions, newAttributeSeq) = c.reusableExpressions()
             // If the input attributes changed, collect current common expressions and clear
             // equivalentExpressions
-            if (!attributeSet.subsetOf(newAttributeSet)) {
+            if (attributeSeq.size != newAttributeSeq.size ||
+              attributeSeq.zip(newAttributeSeq).exists(tup => !tup._1.equals(tup._2))) {
```

Comment:
`tup`?! Please no. How about the following:

```
exists { case (left, right) => left != right }
```

or similar?

I'd also be in favour of creating a helper method to cover this `if` condition 🙏

Reply:
Thanks for your advice.
I want to check if the two attributes seq are the same. I'm not sure if this is a common code?